wampy/transports/websocket/connection.py

```diff
@@ -46,12 +46,13 @@ def connect(self, upgrade=True):
         return self
 
     def disconnect(self):
-        try:
-            self.socket.shutdown(socket.SHUT_RDWR)
-        except socket.error:
```

Comment:
i think it might be better to just 
```
except (socket.error, AttributeError):
```
what do you think? because i reckon `self.socket` can still be `None` under certain conditions.